### PR TITLE
Revert config xds_port

### PR DESCRIPTION
**Pull request by dnephin**
Reverts #10588 and #10618

I learned an upcoming change will actually make use of this port for non-xds services, so the old grpc_port name will end up being correct.

This PR preserves the changes in 4ad80ccee331030c32b669fda9b14d1eacb1628f to use a stable API for the envoy command.

### CODE REVIEW

**Comment by dnephin** [comment]
I think I should keep the fix for the envoy command, I'll restore that.

**Review by karl-cardenas-coding — APPROVED** [review]
👍🏼  from a docs perspective

File: agent/xds/server.go [hunk 1]
```diff
@@ -548,15 +548,14 @@ func tokenFromContext(ctx context.Context) string {
 	return ""
 }
 
-// NewGRPCServer creates a grpc.Server, registers the Server, and then returns
-// the grpc.Server.
-func NewGRPCServer(s *Server, tlsConfigurator *tlsutil.Configurator) *grpc.Server {
```
Comment (freddygv) [inline agent/xds/server.go:553]: Keeping the old name of `NewGRPCServer` could be nice

**Review by freddygv — APPROVED** [review]
LGTM, one minor suggestion below

**Pull request merged**